wolfcrypt/src/wc_port.c

```diff
@@ -1324,7 +1325,17 @@ void wolfSSL_RefInc(wolfSSL_Ref* ref, int* err)
     *err = ret;
 }
 
-void wolfSSL_RefDec(wolfSSL_Ref* ref, int* isZero, int* err)
+int wolfSSL_RefWithMutexLock(wolfSSL_RefWithMutex* ref)
+{
+    return wc_LockMutex(&ref->mutex);
+}
+
+void wolfSSL_RefWithMutexUnlock(wolfSSL_RefWithMutex* ref)
+{
+    wc_UnLockMutex(&ref->mutex);
+}
```

Comment:
Should we log in case of an error in `wc_UnLockMutex`?

Reply:
yes we should -- early detection of state corruption.

did one better and made it return the result, and put the `WOLFSSL_MSG` in `wolfSSL_CTX_UnloadIntermediateCerts()`.